source/lac/trilinos_vector.cc

```diff
@@ -761,33 +761,39 @@ namespace TrilinosWrappers
       AssertThrow (out, ExcIO());
       boost::io::ios_flags_saver restore_flags(out);
 
-      // get a representation of the
-      // vector and loop over all
-      // the elements TODO: up to
-      // now only local data printed
-      // out! Find a way to neatly
-      // output distributed data...
-      TrilinosScalar *val;
-      int leading_dimension;
-      int ierr = vector->ExtractView (&val, &leading_dimension);
 
-      AssertThrow (ierr == 0, ExcTrilinosError(ierr));
       out.precision (precision);
       if (scientific)
         out.setf (std::ios::scientific, std::ios::floatfield);
       else
         out.setf (std::ios::fixed, std::ios::floatfield);
 
-      if (across)
-        for (size_type i=0; i<size(); ++i)
-          out << static_cast<double>(val[i]) << ' ';
+      if (size() != local_size())
+        {
+          auto global_id = [&] (const size_type index)
+          {
+            return vector->Map().GID(static_cast<TrilinosWrappers::types::int_type>(index));
```

Comment:
nice use of a lambda!

Reply:
Is it necessary to distinguish between `GID` and `GID64` in this context?

Reply:
Yeah you probably want to use this function https://github.com/dealii/dealii/blob/master/include/deal.II/lac/trilinos_vector.h#L191